authentication/oauth/oauth.go

```diff
@@ -96,3 +98,10 @@ type RevokeRefreshTokenRequest struct {
 	// Extra parameters to be merged into the request body. Values set here will override any existing values.
 	ExtraParameters map[string]string `json:"-"`
 }
+
+// IDTokenValidationOptionalVerification allows validating optional claims that might not always be in the ID token.
+type IDTokenValidationOptionalVerification struct {
+	MaxAge       time.Duration
+	Nonce        string
+	Organization string
+}
```

Comment:
What about including the clock tolerance here, if it's only used for ID Token validation? Then it would not necessarily be about optional claims.